src/sinks/datadog/metrics/encoder.rs

```diff
@@ -419,6 +455,15 @@ fn sketch_to_proto_message(
     let k = bins.into_iter().map(Into::into).collect();
     let n = counts.into_iter().map(Into::into).collect();
 
+    let event_metadata = metric.metadata();
+    let metadata = generate_sketch_metadata(
+        event_metadata.datadog_origin_metadata(),
+        event_metadata.source_type(),
+        origin_product_value,
+    );
+
+    trace!("Generated sketch metadata: {:?}", metadata);
```

Comment:
Nit: we typically include dynamic bits like `metadata` as key/value rather than string formatting into the log message.